napari/layers/image/image.py

```diff
@@ -546,27 +556,115 @@ def _set_view_slice(self):
             )
             indices[not_disp] = downsampled_indices
 
-            thumbnail_source = np.asarray(
-                self.data[self._thumbnail_level][tuple(indices)]
-            ).transpose(order)
+            thumbnail_source = self.data[self._thumbnail_level][tuple(indices)]
         else:
             self._transforms['tile2data'].scale = np.ones(self.dims.ndim)
-            image = np.asarray(self.data[self.dims.indices]).transpose(order)
-            thumbnail_source = image
+            image_indices = self.dims.indices
+            image = self.data[image_indices]
+
+            # For single-scale we don't request a separate thumbnail_source
+            # from the ChunkLoader because in ImageSlice.chunk_loaded we
+            # call request.thumbnail_source() and it knows to just use the
+            # image itself is there is no explicit thumbnail_source.
+            thumbnail_source = None
+
+        # Load our images, might be sync or async.
+        key = ChunkKey(self, image_indices)
+        self._load_images(key, image, thumbnail_source)
+
+    def _load_images(self, key: ChunkKey, image, thumbnail_source=None):
+        """Load the image and maybe thumbnail source.
+
+        The load will happen synchronously if any of these are true:
+        1) The arrays are already ndarrays, or
+        2) We are in synchronous mode, or
+        3) The arrays were found in the ChunkCache.
+
+        If the load is synchronous we immediately call our chunk_loaded()
+        method so that we use the new data right away. If the load is
+        asynchronous then sometime later our chunk_loaded() method will be
+        called with the loaded data.
+        """
+        # We always load the image.
+        chunks = {'image': image}
+
+        # Optionally also load the thumbnail_source.
+        if thumbnail_source is not None:
+            chunks['thumbnail_source'] = thumbnail_source
+
+        # Create the ChunkRequest for the chunks we are going to load.
+        request = chunk_loader.create_request(self, key, chunks)
 
-        if self.rgb and image.dtype.kind == 'f':
-            self._slice.image.raw = np.clip(image, 0, 1)
-            self._slice.thumbnail.raw = np.clip(thumbnail_source, 0, 1)
+        # Load the chunk(s), the load could happen sync or async.
+        satisfied_request = self._slice.load_chunk(request)
+
+        if satisfied_request is None:
+            # The load is going to be async, announce that a load is
+            # underway, that we are no longer in the loaded state.
```

Comment:
I think the two comments are consistent. But perhaps they are both confusing since "load" is used over and over. I would try to explain it like this:

The `Image.loaded` property should be true iff "the layer has all the data it needs to draw itself". 

The cycle is meant to be:
* `Image` is not loaded, because its `ImageSlice` is not loaded
* `Image._load_images` -> `ImageSlice.load_chunk` -> `chunk_loader.load_chunk`
* If that load is synchronous:
    * `Image.chunk_loaded` -> `ImageSlice.chunk_loaded`
    * `ImageSlice.loaded` is now true, so `Image.loaded` is true
    * Fire event `Image.events.loaded`
    * VispyBaseLayer received `loaded` event and turns the node visible
* If that load is asynchronous `Image.loaded` stays false
   * Sometime later `Image.chunk_loaded` is called same as above.

This kind of lifecycle diagram probably needs to be documented somewhere outside the code, although some big centralized single-comment in the code might make sense too.

Reply:
thanks for the explanation.  please put in comments somewhere

Reply:
👍 on @tlambert03's suggestion: regardless of a documenting diagram (which is also a good idea), having this comment next to the relevant parts of the code is super handy.